client/src/pages/Analytics/components/CategoryGraph.tsx

```diff
@@ -17,24 +21,85 @@ const categoryTotals = [
  * @returns {JSX.Element} - category graph
  */
 export const CategoryGraph: React.FC = () => {
+  const { userId, token } = useUser();
+  const [categoryTotals, setCategoryTotals] = React.useState<
+    { id: number; value: number; label: string }[]
+  >([]);
+  const [errorState, setErrorState] = React.useState(false);
+  const currDate = new Date();
+  const startDate = new Date(currDate.getFullYear(), currDate.getMonth(), 1);
+
+  React.useEffect(() => {
+    fetchCategoryExpenses(startDate.toISOString(), currDate.toISOString());
+  }, []);
+
+  const fetchCategoryExpenses = async (startDate: string, endDate: string) => {
+    try {
+      if (userId != null && token != null) {
+        const response = await categoryTotalExpensesRequest(
+          { userId, token },
+          { startDate, endDate },
+        );
+
+        if (response.status === 200) {
+          const totals: { category: string; amount: number }[] =
+            response.data.data;
+          const processedTotals = processData(totals);
+          setCategoryTotals(processedTotals);
+        }
+      }
+    } catch (error) {
+      setErrorState(true);
+    }
+  };
+
+  const processData = (
+    totals: { category: string; amount: number }[],
+  ): { id: number; value: number; label: string }[] => {
+    const categoryMap = new Map(
+      totals.map((item) => [item.category, item.amount]),
+    );
+    const processedData = allCategories.map((category, index) => ({
+      id: index,
+      value: categoryMap.get(category) || 0,
+      label: category,
+    }));
+
+    return processedData;
+  };
+
   return (
     <BackgroundBox>
-      <BarChart
-        xAxis={[
-          {
-            dataKey: 'category',
-            scaleType: 'band',
-          },
-        ]}
-        series={[
-          {
-            dataKey: 'amount',
-            label: 'Total Expense',
-            color: '#85D3C3',
-          },
-        ]}
-        dataset={categoryTotals}
-      />
+      {!errorState && categoryTotals.length != 0 && (
+        <PieChart
+          colors={[
+            '#FFB7B2',
+            '#B2FFCC',
+            '#EBB2FF',
+            '#FFF3B2',
+            '#FFB2E5',
+            '#B2C3FF',
+          ]}
```

Comment:
could you add the whole colours array to the constanst or name each one, just for readability